src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/DirectiveAttributeTransitionCompletionItemProvider.cs

```diff
@@ -53,25 +53,56 @@ public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorSynta
                 return Completions;
             }
 
-            if (!TryGetAttributeInfo(owner, out var prefixLocation, out var name, out var nameLocation, out _, out _))
+            if (!TryGetAttributeInfo(owner, out var prefixLocation, out var attributeName, out var attributeNameLocation, out _, out _))
             {
                 return Array.Empty<RazorCompletionItem>();
             }
 
-            if (nameLocation.IntersectsWith(location.AbsoluteIndex) && name.StartsWith("@"))
+            if (attributeNameLocation.IntersectsWith(location.AbsoluteIndex) && attributeName.StartsWith("@"))
             {
-                // The transition is already provided for the name
+                // The transition is already provided for the attribute name
                 return Array.Empty<RazorCompletionItem>();
             }
 
-            if (!IntersectsWithAttributeNameOrPrefix(location, prefixLocation, nameLocation))
+            if (!IsValidCompletionPoint(location, prefixLocation, attributeNameLocation))
             {
-                // Not operating in the name area
+                // Not operating in the attribute name area
                 return Array.Empty<RazorCompletionItem>();
             }
 
             // This represents a tag when there's no attribute content <InputText | />.
             return Completions;
         }
+
+        // Internal for testing
+        internal static bool IsValidCompletionPoint(SourceSpan location, TextSpan? prefixLocation, TextSpan attributeNameLocation)
+        {
+            if (location.AbsoluteIndex == (prefixLocation?.Start ?? -1))
+            {
+                // <input| class="test" />
+                // Starts of prefix locations belong to the previous SyntaxNode. It could be the end of an attribute value, the tag name, C# etc.
+                return false;
+            }
+
+            if (attributeNameLocation.Start == location.AbsoluteIndex)
+            {
+                // <input |class="test" />
+                return false;
+            }
+
+            if (prefixLocation?.IntersectsWith(location.AbsoluteIndex) ?? false)
+            {
+                // <input   |  class="test" />
+                return true;
+            }
+
+            if (attributeNameLocation.IntersectsWith(location.AbsoluteIndex))
+            {
+                // <input cla|ss="test" />
+                return false;
```

Comment:
This is a change in behavior, we were previously providing completions in this context. Per the initial issue:

> We should only be providing these attribute completions if there's whitespace prior to the cursor location and some sort of delimiter after it. i.e. > any sort of whitespace etc.